### PR TITLE
feat(sast): Download and run prisma engine

### PR DESCRIPTION
**Pull request by matansha**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/sast/engines/prisma_engine.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            return []
+
+        self.setup_sast_artifact()
+        prisma_lib_path = self.get_sast_artifact()
+        if not prisma_lib_path:
+            return []
+
+        self.lib_path = str(prisma_lib_path)
+
+        prisma_resut = self.run_go_library(languages, source_codes=targets, policies=registry.checks_dirs_path)
```
Comment (achiar99) [inline checkov/sast/engines/prisma_engine.py:53]:
```suggestion
        prisma_result = self.run_go_library(languages, source_codes=targets, policies=registry.checks_dirs_path)
```

File: Pipfile [hunk 2]
```diff
@@ -87,6 +87,7 @@ schema = "*"
 requests = ">=2.27.0"
 yarl = "*"
 semgrep = "==1.10.0"
+pydantic = "==1.10.7"
```
Comment (achiar99) [inline Pipfile:90]: 🌟

**Review by achiar99 — APPROVED** [review]
🏆

File: checkov/cdk/checks_infra/base_registry.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ def __init__(
         self, checks_dir: str, temp_semgrep_rules_path: str | None = None
     ) -> None:
         super().__init__(checks_dir=checks_dir, temp_semgrep_rules_path=temp_semgrep_rules_path)
-        self.report_type = CheckType.CDK
+        self.report_type = CheckType.SAST
```
Comment (gruebel) [inline checkov/cdk/checks_infra/base_registry.py:15]: the `report_type` should stay as `CDK`

File: checkov/sast/checks_infra/base_registry.py [hunk 4]
```diff
@@ -33,6 +37,22 @@ def extract_entity_details(self, entity: dict[str, Any]) -> tuple[str, str, dict
     def set_runner_filter(self, runner_filter: RunnerFilter) -> None:
         self.runner_filter = runner_filter
 
+    def add_external_dirs(self, external_dirs: Optional[List[str]]) -> None:
+        if external_dirs and len(external_dirs) > 0:
```
Comment (rotemavni) [inline checkov/sast/checks_infra/base_registry.py:41]: Do we really need the added `len(external_dirs) > 0`? Isn't `if external_dirs` enough?

File: checkov/cdk/checks_infra/base_registry.py [hunk 5]
```diff
@@ -18,3 +18,9 @@ def load_rules(
         self, frameworks: Iterable[str], sast_languages: set[SastLanguages] | None
     ) -> int:
         return self._load_checks_from_dir(directory=self.checks_dir, sast_languages=SastLanguages.set())
+
+    def load_semgrep_checks(self, languages: set[SastLanguages]) -> int:
```
Comment (rotemavni) [inline checkov/cdk/checks_infra/base_registry.py:22]: Is this a duplication of the method in `checkov/sast/checks_infra/base_registry.py`? Do we need them both?

File: checkov/sast/engines/prisma_engine.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+                              (self.prisma_sast_dir_path / f).is_file() and "library" in f]
+            if len(is_file_exists) > 0:
+                match = re.match(r"(\d+_\d+_\d+)_library\.(so|dll)", is_file_exists[0])
+                if match:
+                    current_version = match.groups()[0]
+
+        status: bool = self.download_sast_artifacts(current_version)
+        return status
+
+    def download_sast_artifacts(self, current_version: str) -> bool:
```
Comment (rotemavni) [inline checkov/sast/engines/prisma_engine.py:70]: How do you feel about adding some kind of caching here like `@cached(TTLCache(maxsize=1, ttl=540))`?

**Review by rotemavni — APPROVED** [review]

File: checkov/cdk/checks_infra/base_registry.py (hunk 5 above)
Comment (matansha) [reply checkov/cdk/checks_infra/base_registry.py:22]: yes we do, this is the only way that unitest work's. the logic inside changed

File: checkov/sast/checks_infra/base_registry.py (hunk 4 above)
Comment (matansha) [reply checkov/sast/checks_infra/base_registry.py:41]: right 😸

File: checkov/sast/engines/prisma_engine.py (hunk 6 above)
Comment (matansha) [reply checkov/sast/engines/prisma_engine.py:70]: change it to 5m

**Pull request merged**